File: apply_fix.py
#!/usr/bin/env python3
import sys

# Read the file
with open('src/components/WarehouseSpecification.tsx', 'r') as f:
    content = f.read()

# Find the pattern and replace it
old_code = '''    setExpandedItems(updatedItems);
    
    // Mark parent item as having applied modifications
    setItemsWithAppliedModifications(prev => {
      const newSet = new Set(prev);
      newSet.add(selectedBudgetItemForMod.id);
      return newSet;
    });'''

new_code = '''    setExpandedItems(updatedItems);

    // Mark the new modification component items as modified so they get saved
    setModifiedItems(prev => {
      const newSet = new Set(prev);
      newItems.forEach(item => newSet.add(item.budgetItemId));
      return newSet;
    });

    // Mark parent item as having applied modifications
    setItemsWithAppliedModifications(prev => {
      const newSet = new Set(prev);
      newSet.add(selectedBudgetItemForMod.id);
      return newSet;
    });'''

if old_code in content:
    updated_content = content.replace(old_code, new_code)
    with open('src/components/WarehouseSpecification.tsx', 'w') as f:
        f.write(updated_content)
    print("✓ Fix successfully applied!")
    sys.exit(0)
else:
    print("✗ Pattern not found - file may already be fixed")
    sys.exit(1)

File: apply_fix.js
const fs = require('fs');

const filePath = 'src/components/WarehouseSpecification.tsx';
let content = fs.readFileSync(filePath, 'utf8');

const oldCode = `    setExpandedItems(updatedItems);
    
    // Mark parent item as having applied modifications
    setItemsWithAppliedModifications(prev => {
      const newSet = new Set(prev);
      newSet.add(selectedBudgetItemForMod.id);
      return newSet;
    });`;

const newCode = `    setExpandedItems(updatedItems);

    // Mark the new modification component items as modified so they get saved
    setModifiedItems(prev => {
      const newSet = new Set(prev);
      newItems.forEach(item => newSet.add(item.budgetItemId));
      return newSet;
    });

    // Mark parent item as having applied modifications
    setItemsWithAppliedModifications(prev => {
      const newSet = new Set(prev);
      newSet.add(selectedBudgetItemForMod.id);
      return newSet;
    });`;

if (content.includes(oldCode)) {
  content = content.replace(oldCode, newCode);
  fs.writeFileSync(filePath, content);
  console.log('✓ Fix successfully applied!');
} else {
  console.log('✗ Pattern not found - file may already be fixed');
}
